tests/smoke/functionality.tests.js

```diff
@@ -0,0 +1,143 @@
+const { setting } = require('../../src/index');
+
+var homeScreenFunctionalityTests = (test) => {
+    it('has Active Carbohydrates section', async () => {
+        await test.homeScreen.OpenActiveCarbohydratesChart();
+        await test.homeScreen.CloseChart();
+    });
+    it('has Active Insulin section', async () => {
+        await test.homeScreen.OpenActiveInsulinChart();
+        await test.homeScreen.CloseChart();
+    });
+    it('has Insulin Delivery section', async () => {
+        await test.homeScreen.OpenInsulinDeliveryChart();
+        await test.homeScreen.CloseChart();
+    });
+    it('has Glucose section', async () => {
+        await test.homeScreen.OpenGlucoseChart();
+        await test.homeScreen.CloseChart();
+    });
+    it('has Loop icon', async () => {
+        await test.homeScreen.ExpectLoopNotYetRun();
+    });
+    it.skip('has Loop icon has alert when not setup', async () => {
+        await test.homeScreen.ExpectLoopStatusGlucoseDataAlert();
+    });
+};
+
+var settingsScreenFunctionalityTests = (test) => {
+    it('can open the settings', async () => {
+        await test.settingsScreen.Open();
+    });
+    describe('general', () => {
+        it('set to closed loop', async () => {
+            await test.settingsScreen.SetClosedLoop();
+        });
+        it('set to open loop', async () => {
+            await test.settingsScreen.SetOpenLoop();
+        });
+        it('open issue report', async () => {
+            await test.settingsScreen.OpenIssueReport();
+        });
+        it('close issue report', async () => {
+            await test.settingsScreen.CloseIssueReport();
+        });
+    });
+    describe('cgm', () => {
+        it('can be added', async () => {
+            await test.settingsScreen.AddCGMSimulator();
+        });
+        it('can configure simulator', async () => {
+            let screen = await test.settingsScreen.OpenCGMSimulatorScreen();
+            await screen.Apply(setting.default.CGMSimulatorSettings);
+            await screen.Close();
+        });
+    });
+    describe('pump', () => {
+        it('can be added', async () => {
+            await test.settingsScreen.AddPumpSimulator();
+        });
+        it.skip('set suspend threshold', async () => {
```

Comment:
[nit] At some point might want to do a global find for all "skip" and fix anything broken.